dspace-spring-rest/src/test/java/org/dspace/app/rest/CollectionRestRepositoryIT.java

```diff
@@ -624,6 +626,28 @@ public void updateCollectionEpersonWithWriteRightsTest() throws Exception {
         ;
 
         authorizeService.removePoliciesActionFilter(context, eperson, Constants.WRITE);
+    }
+
+    public void patchCollectionMetadataAuthorized() throws Exception {
+        runPatchMetadataTests(admin, 200);
+    }
 
+    @Test
+    public void patchCollectionMetadataUnauthorized() throws Exception {
+        runPatchMetadataTests(eperson, 403);
+    }
+
+    private void runPatchMetadataTests(EPerson asUser, int expectedStatus) throws Exception {
+        context.turnOffAuthorisationSystem();
+        parentCommunity = CommunityBuilder.createCommunity(context).withName("Community").build();
+        Collection col = CollectionBuilder.createCollection(context, parentCommunity).withName("Collection").build();
+        String token = getAuthToken(asUser.getEmail(), password);
+
+        try {
+            new MetadataPatchSuite().runWith(getClient(token), "/api/core/collections/"
+                    + col.getID(), expectedStatus);
+        } finally {
+            context.restoreAuthSystemState();
```

Comment:
@cwilper : could you explain your reasoning for putting the `context.restoreAuthSystemState()` in a `finally`?  (in one of your most recent commits) 

To clarify my earlier request, I was wanting to see the `context.restoreAuthSystemState()` called **before** any calls to the REST API are made.  Context authorization is only needed to be turned off to create test content.  After test content is created, we should call that `restoreAuthSystemState()`.  

While this Context object shouldn't end up having an effect on the REST API call (in the `try`), it's a bit disconcerting to have a REST API test that is specifically testing proper authorization access rights alongside a Context object that has all authorization checks *disabled*.  

In all tests, I think we need to keep authorization rights enabled **for as much of the test as possible**...otherwise I worry we'll eventually encounter unexpected behavior in tests (maybe not in this specific test, but perhaps in another that copies & pastes this same logic later on).

Reply:
Long story short, I think the `context.restoreAuthSystemState();` call should be moved to just *after* the test content is created (in all methods).

Reply:
@tdonohue Oops, you're right -- these restores should definitely happen earlier. I read the code too quick. Will correct to restore stats as soon as possible, and not have it operating in the non-enforcing way for the actual test -- just the test setup.